symengine/functions.cpp

```diff
@@ -327,8 +327,37 @@ RCP<const Basic> sin(const RCP<const Basic> &arg)
     if (is_a_Number(*arg) and not static_cast<const Number &>(*arg).is_exact()) {
         return static_cast<const Number &>(*arg).get_eval().sin(*arg);
     }
-    if (is_a<ASin>(*arg)) {
-        return rcp_static_cast<const ASin>(arg)->get_arg();
+
+    RCP<const Basic> i_arg;
+    if (is_a<ASin>(*arg)) 
```

Comment:
You can introduce a new function `is_a_TrigFunction` which should be similar to `is_a_Number` and then `i_arg` would be `rcp_static_cast<const TrigFunction>(arg)->get_arg();` or simply `static_cast<const TrigFunction &>(arg).get_arg();`